### PR TITLE
Disabling test failing in Mac OSX 10.15

### PR DESCRIPTION
**Pull request by buyaa-n**
Mac OSX 10.15 not triggering Create event anymore for `target` path for `File.Move(source, target)` action, this need to investigated more and handled, the issue tracked by #40034  meanwhile disabling test for OSX.

### CODE REVIEW

File: src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Move.cs [hunk 1]
```diff
@@ -50,6 +50,7 @@ public void Linux_File_Move_To_Different_Watched_Directory()
         }
 
         [Fact]
+        [PlatformSpecific(~TestPlatforms.OSX)] // Not supported since OSX 10.15 Issue: https://github.com/dotnet/corefx/issues/40034
```
Comment (stephentoub) [inline src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Move.cs]: Can you use `[ActiveIssue(40034, TestPlatforms.OSX)]` instead?

Comment (buyaa-n) [reply src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Move.cs]: Sure

**Comment by buyaa-n** [comment]
@eerhardt for your suggestion on scru:  `to check OS version instead of disabling the test` I think its better to keep it as is with following reason:
1. Keep active issue for tracking and enable test after its fixed
2. From what am seeing from CI builds we are testing only for one version of Mac which I guess the latest version, version based conditional test not sounds so beneficial, especially in case OS version upgraded in the future

Please let me know if you ok with leaving it as is or still prefer to have condition within the test

**Comment by eerhardt** [comment]
From looking at our pipelines for mac:

https://github.com/dotnet/corefx/blob/ba54c4034a0e87ac6bdf3d13b55b767f5c321df2/eng/pipelines/macos.yml#L58-L63

We are running tests on `macOS 10.13` and `macOS 10.14`.

Are we concerned that we will be losing coverage here by disabling this test on all macOS versions? The test will pass on `10.13` and `10.14`. So turning it off on all macOS versions, just because it fails on `10.15` seems wrong to me.

I agree an `ActiveIssue` is best, but we have the issue in GitHub to do the tracking.

**Comment by buyaa-n** [comment]
Ok then will add condition for OS version check

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**